pkg/codegen/python/gen.go

```diff
@@ -791,9 +790,6 @@ func (mod *modContext) importObjectType(t *schema.ObjectType, input bool) string
 	}
 
 	importPath := mod.getRelImportFromRoot()
-	if mod.pkg.Name() != parts[0] {
```

Comment:
Note that we handle external imports at the start of the method:

https://github.com/pulumi/pulumi/blob/44f20644839943e0a9bc300213ee98d3e9aa6112/pkg/codegen/python/gen.go#L776-L778

What I'm removing seems to be leftover from a previous attempt to support this, which is causing the problems.